vowpalwabbit/parse_args.cc

```diff
@@ -1070,6 +1070,29 @@ void parse_example_tweaks(options_i& options, VW::workspace& all)
   }
 }
 
+void parse_update_options(options_i& options, VW::workspace& all)
+{
+  option_group_definition update_args("Update");
+  update_args
+      .add(make_option("learning_rate", all.eta)
+               .default_value(0.5f)
+               .keep()
+               .allow_override()
+               .help("Set learning rate")
+               .short_name("l"))
+      .add(make_option("power_t", all.power_t).default_value(0.5f).keep().allow_override().help("T power value"))
+      .add(make_option("decay_learning_rate", all.eta_decay_rate)
+               .default_value(1.f)
+               .help("Set Decay factor for learning_rate between passes"))
+      .add(make_option("initial_t", all.sd->t).keep().allow_override().help("Initial t value"))
```

Comment:
Why is this needed as keep. Both t and initial_t are serialized as part of a save_resume model

Reply:
I dont think being part of save_resume is sufficient here. If you save and load a model without `keep()`, the output will print `initial_t` with the default value (although internally initial_t and sd->t should restore properly)

Reply:
save_resume isn't sufficient here because it is called after the initial output printing, so if you save and load a model with some `initial_t` it will print the default value (although internally it should restore initial_t at a later point)

Reply:
But now dont we have duplicated/potentially conflicting initial_t information between the command line and the model?

Reply:
I removed `keep()` and `allow_override()` from `initial_t`, and moved the printing to after the model is loaded. This is actually a bug fix since `initial_t` is often altered before being save_loaded, and was previously printing out the default for loaded models.